### ISSUE TITLE
Why not show conn string by default in iothub-explorer get?

### ISSUE THREAD

**Issue by snobu** [Azure/iothub-explorer#13]
Since we already show the secrets in the console, why not include the Connection String by default in the same output?

This would add to the overall user experience. Currently, you do

```bash
% iothub-explorer get DEVICE_ID
```

look for the connection string, see it's not there, you think _Hmm maybe it's not in the `get` context_, then you do this

```bash
% iothub-explorer | grep -i conn
%
```

get confused, Google for `iothub-explorer connection string`, end up on the README here and realize you have to pass `--connection-string` to `get`, feel a bit silly and to make things better you open an issue.

This is not easily discoverable in my opinion:
```bash
% iothub-explorer get -h
  ...
  Options:

    -h, --help                       output usage information
    -c, --connection-string          show the connection string for the newly
                                     created device
```

**Comment by pierreca** [comment]
That's good feedback, thanks. 

I think the original idea was to separate what is stored in the device registry (`deviceId`, `symmetricKeys`...) from what is/should be generated by the user (the connection string). Then we changed the way the `get` command work and integrated the connection string as part of the description, although it's still not part of what is stored in the registry. The idea was to make scripting easier by formatting everything as JSON instead of 2 separate output formats for the same command. In hindsight I understand this could even cause more confusion. 

I guess there are multiple ways to solve this, I'd be curious to understand which one you think is better:
- keep everything formatted as JSON and add the connection string, all the time, making the `--connection-string` option useless _(you seem to be in favor of that one)_
- return to the previous format (JSON for the device registry content, plus an additional, separate line for the connection string), except print the connection string separately from the JSON to clarify that this is not a part of what is stored in the registry. (would be a breaking change though).
- assume that the only thing that is desired when using `--connection-string` is the connection string and print only this (and not the whole JSON description). That would feel counterproductive with regards to what you're asking but I can see how it would make sense from a scripting perspective too.

**Comment by snobu** [comment]
Ha! I actually just realized that the Connection String is something the client constructs based on strings it gets from the registry. I guess i got too used to Device Explorer always handing out the Connection String for me. Differentiating output now starts to make much more sense.

That being said, for folks like me that aren't very clear on the mechanics, it would still be beneficial to add the Connection String to the output of both `get` and `list`.

I choose door number #1 (make the `--connection-string` parameter a dummy one and always include it in the JSON object). I'd go even further and add a mention of the `connectionString` not being part of the registry right in the usage listing. Something along the lines of:

```bash
% iothub-explorer                                                                                                         ~

  Usage: iothub-explorer [options] <command> [command-options] [command-args]

  Commands:

    login                                                                          start a session on your IoT hub
    logout                                                                         terminate the current session on your IoT hub
    list                                                                           list the device identities currently in your IoT hub device registry
[...]
  Options:

    -h, --help     output usage information
    -V, --version  output the version number

The connectionString property present in the get and list output is not stored in the registry but constructed on the fly for both discoverability and convenience. To learn more visit this page: http://<link to a markdown page explaining the internals>
```

This way we don't break anyone and we also give (what i believe to be) important, easily discoverable information.

**Comment by pierreca** [comment]
Looks like the team agrees with this approach. We'll have that change checked-in shortly and then we'll release a new package. Again, thanks a lot for the very good feedback.

**Comment by pierreca** [comment]
This should do: https://github.com/Azure/iothub-explorer/commit/d5e8d0bdf2660ef1744c41a96a5e9274c7e5acd9

We'll release a new package quickly.

**Comment by pierreca** [comment]
Fix has been released with iothub-explorer@1.1.9. Thanks!

**Comment by snobu** [comment]
This is awesome you guys!

![image](https://cloud.githubusercontent.com/assets/6472374/23559382/774b1b2c-003f-11e7-82f2-34faaae72f9a.png)
